### ISSUE TITLE
Low latency modes only for recording

### ISSUE THREAD

**Issue by trillerr** [Xaymar/OBS-AMD-Advanced-Media-Framework#182]
I am curious why low latency / ultra low latency modes are only available for recording.  Is this a limitation of obs, amf, or the plugin?  I was digging through your source and the encoding api and wasn't able to find anything specific enforcing this.  The encoding api does say ultra low latency is intended for video game streaming :)

**Comment by Xaymar** [comment]
It's meant for streaming, but not that kind of streaming it seems like. Twitch, YouTube, HitBox, etc will all disconnect you with low latency presets. You can use them for recording, but be aware of the extreme impact on quality.

**Comment by trillerr** [comment]
So I am trying to stream over local network but when in low latency modes the encoder does not start.  Is this likely related to other settings, or is it actually disabled?  I can't recall specifically what errors I was seeing in the log.  I will check those once I am home this evening.

**Comment by Xaymar** [comment]
Take a look at the log file, it should give you an idea which one of your settings is incompatible.

**Comment by trillerr** [comment]
Here is a log.  Looks like the rate control mode fails in ultra low latency regardless of what mode is selected.  
[2017-01-04 19-13-18.txt](https://github.com/Xaymar/obs-studio_amf-encoder-plugin/files/686228/2017-01-04.19-13-18.txt)

Here is a log with low latency.  This shows B-frames being an issue (among other things).  As I am not enabling B frames I am guessing this is due to the hidden defaults.
[2017-01-04 19-22-03.txt](https://github.com/Xaymar/obs-studio_amf-encoder-plugin/files/686234/2017-01-04.19-22-03.txt)

Before running I used the reset to defaults then changed to speed preset for ultra low and nothing on the low latency runs.

**Comment by Xaymar** [comment]
Trying to fix it without breaking other functionality. I hope it will work, but it is an experimental/master property after all.

**Comment by Xaymar** [comment]
Changes necessary for this are now implemented, here's a test:

### Transcoding PerfMon
```
01:06:16.328: [AMF Encoder] [01:06:16.253][PerfMon]     Info: ------ Performance results for AMFEncoderVCEImpl:
01:06:16.328: [AMF Encoder] [01:06:16.253][PerfMon]     Info: Metric              CreateEncoder                   SetEncodeConfig                 SubmitInput                     CheckAndUpdateConfig            CheckResoultion                 BufferReadParams                AttachAmfBuffer                 AVEEncode                       BufferQueue_Add                 QueryOutput                     BufferQueue_Pop                 
01:06:16.328: [AMF Encoder] [01:06:16.253][PerfMon]     Info: Average/Call (msec) 1.458                           0.276                           0.280                           0.000                           0.000                           0.023                           0.001                           0.086                           0.012                           0.158                           0.024                           
01:06:16.328: [AMF Encoder] [01:06:16.253][PerfMon]     Info: Min  (msec)         1.458                           0.276                           0.144                           0.000                           0.000                           0.014                           0.000                           0.050                           0.007                           0.055                           0.014                           
01:06:16.328: [AMF Encoder] [01:06:16.253][PerfMon]     Info: Max  (msec)         1.458                           0.276                           0.568                           0.037                           0.016                           0.222                           0.014                           0.437                           0.060                           0.389                           0.095                           
01:06:16.329: [AMF Encoder] [01:06:16.253][PerfMon]     Info: Total (msec)        1.458                           0.276                           172.082                         0.293                           0.082                           14.430                          0.629                           52.876                          7.217                           97.310                          14.803                          
01:06:16.329: [AMF Encoder] [01:06:16.253][PerfMon]     Info: Calls number        1                               1                               615                             615                             615                             615                             615                             615                             615                             615                             613                             
```

### Ultra Low Latency PerfMon
```
01:07:36.126: [AMF Encoder] [01:07:36.057][PerfMon]     Info: ------ Performance results for AMFEncoderVCEImpl:
01:07:36.126: [AMF Encoder] [01:07:36.057][PerfMon]     Info: Metric              CreateEncoder                   SetEncodeConfig                 SubmitInput                     CheckAndUpdateConfig            CheckResoultion                 BufferReadParams                AttachAmfBuffer                 AVEEncode                       BufferQueue_Add                 QueryOutput                     BufferQueue_Pop                 
01:07:36.126: [AMF Encoder] [01:07:36.057][PerfMon]     Info: Average/Call (msec) 4.846                           0.269                           0.260                           0.001                           0.000                           0.021                           0.001                           0.080                           0.011                           0.160                           0.024                           
01:07:36.126: [AMF Encoder] [01:07:36.057][PerfMon]     Info: Min  (msec)         4.846                           0.269                           0.121                           0.000                           0.000                           0.014                           0.000                           0.048                           0.007                           0.055                           0.016                           
01:07:36.126: [AMF Encoder] [01:07:36.057][PerfMon]     Info: Max  (msec)         4.846                           0.269                           0.863                           0.026                           0.001                           0.082                           0.038                           0.805                           0.055                           0.910                           0.266                           
01:07:36.126: [AMF Encoder] [01:07:36.057][PerfMon]     Info: Total (msec)        4.846                           0.269                           313.902                         0.784                           0.016                           24.805                          1.278                           96.151                          12.847                          193.842                         28.466                          
01:07:36.126: [AMF Encoder] [01:07:36.057][PerfMon]     Info: Calls number        1                               1                               1208                            1208                            1208                            1208                            1208                            1208                            1208                            1208                            1206                            
```

Transcoding seems to overall produce much more stable call times and also has a much lower max call time. Also seeking with a Low Latency file seems to be kind of broken unless you turn off everything else (including full range encoding). I wouldn't recommend it for in-home streaming, but it is a possibility. If you really need those few microseconds on average

**Comment by Xaymar** [comment]
Fully implemented now, will be in 1.4.3.5.
